src/test/java/com/ontotext/kafka/service/AddRecordsProcessorTest.java

```diff
@@ -134,11 +135,187 @@ void testWriteBatchTimeoutAfterProperBatching() throws InterruptedException, IOE
+		int batch = 4;
+		int expectedSize = 1;
+
+		repository = initThrowingRepository(streams, formats, new RDFParseException("RDFParseException"), 3);
+		generateSinkRecords(sinkRecords, 4, 12);
+		Thread recordsProcessor = createProcessorThread(sinkRecords, shouldRun, repository, batch, 50);
+		recordsProcessor.start();
+		awaitEmptyCollection(sinkRecords);
+		shouldRun.set(false);
+		awaitProcessorShutdown(recordsProcessor);
+
+		assertEquals(expectedSize, streams.size());
+		for (Reader reader : streams) {
+			assertEquals(12, Rio.parse(reader, RDFFormat.NQUADS).size());
+		}
+	}
+
+	@Test
+	@DisplayName("Test all records are malformed RDFParseException")
+	@Timeout(5)
+	void testHandleAllRDFParseExceptionRecords() throws InterruptedException {
+		int batch = 4;
+		int expectedSize = 0;
+
+		repository = initThrowingRepository(streams, formats, new RDFParseException("RDFParseException"), 4);
+		generateSinkRecords(sinkRecords, 4, 12);
+		Thread recordsProcessor = createProcessorThread(sinkRecords, shouldRun, repository, batch, 50);
+		recordsProcessor.start();
+		awaitEmptyCollection(sinkRecords);
+		shouldRun.set(false);
+		awaitProcessorShutdown(recordsProcessor);
+
+		assertEquals(expectedSize, streams.size());
+	}
+
+	@Test
+	@DisplayName("Test RepositoryException should get handled")
+	@Timeout(5)
+	void testHandleRepositoryException() throws InterruptedException, IOException {
+		int batch = 4;
+		int expectedSize = 3;
+
+		repository = initThrowingRepository(streams, formats, new RepositoryException());
+		generateSinkRecords(sinkRecords, 4, 12);
+		Thread recordsProcessor = createProcessorThread(sinkRecords, shouldRun, repository, batch, 50);
+		recordsProcessor.start();
+		awaitEmptyCollection(sinkRecords);
+		awaitCollectionSizeReached(streams, expectedSize);
+		shouldRun.set(false);
+		awaitProcessorShutdown(recordsProcessor);
+
+		assertEquals(expectedSize, streams.size());
+		for (Reader reader : streams) {
+			assertEquals(12, Rio.parse(reader, RDFFormat.NQUADS).size());
+		}
+	}
+
+	@Test
+	@DisplayName("Test IOExceptions should get retried")
+	@Timeout(5)
+	void testHandleIOException() throws InterruptedException, IOException {
+		int batch = 4;
+		int expectedSize = 4;
+
+		repository = initThrowingRepository(streams, formats, new IOException());
+		generateSinkRecords(sinkRecords, 4, 12);
+		Thread recordsProcessor = createProcessorThread(sinkRecords, shouldRun, repository, batch, 50);
+		recordsProcessor.start();
+		awaitEmptyCollection(sinkRecords);
+		awaitCollectionSizeReached(streams, expectedSize);
+		shouldRun.set(false);
+		awaitProcessorShutdown(recordsProcessor);
+
+		assertEquals(expectedSize, streams.size());
+		for (Reader reader : streams) {
+			assertEquals(12, Rio.parse(reader, RDFFormat.NQUADS).size());
+		}
+	}
+
+	@Test
+	@Disabled//it is slow
+	@DisplayName("Test should throw when out of retries")
+	@Timeout(35)
+	void testThrowTimoutExceptionException() {
+		int batch = 1;
+
```

Comment:
why is this test slow?

Reply:
because by default it will do 300 retries at rate 10/s ~ 30 sec